### PR TITLE
Visually more uniform colortable

### PR DESCRIPTION
**Pull request by timhoffm**
High variations in saturation make the colortable somewhat non-uniform. By splitting into low/high saturation groups, the table IMHO looks better.

image

### CODE REVIEW

**Comment by jklymak** [comment]
.. should add, I don't see why we couldn't include both sortings, if both seem useful.

**Comment by tacaswell** [comment]
If we are doing multiple sortings we should also include the alphabetical one as well.

**Comment by timhoffm** [comment]
** TL;DR: Let's table this for later.**

Still, I want to comment on your remarks.

> by-hue makes more sense to me than by-saturation. Usually you are looking for a shade of green, not any color with a certain saturation.

I believe a fundamental partition by-saturation makes sense. Low saturation is more suited for large areas (facecolor), high saturation is more suited for small objects (text, marker, ...). Note that for example http://davidbau.com/archives/2013/09/28/css_color_names.html also aggregates low-saturation colors together.

> Like I really don't like splitting the grays

That [has always been there](https://matplotlib.org/devdocs/gallery/color/named_colors.html), but I agree it should be improved.

> I'm partial to sorting on hues cause I'm usually looking at that table to find/build a gradient?

I would say, that the old one is not at all better for this:

old: image new: image

The strong variation on saturation makes the gradient uneven.

**Comment by timhoffm** [comment]
You could try, but google has the advantage that they can choose the colors, and thus choose a grid in (hue, saturation) space.

The CSS4 colors are unevenly spaced, so that it may be quite hard pressing them in a (hue, saturation) grid:

```
import matplotlib.pyplot as plt
import matplotlib.colors as mcolors
hsv = np.array([mcolors.rgb_to_hsv(mcolors.to_rgb(c)) for c in mcolors.CSS4_COLORS])
plt.scatter(hsv[:,0], hsv[:,1], c=mcolors.CSS4_COLORS)
plt.xlabel("hue")
plt.ylabel("sat")
```

image

**Comment by timhoffm** [comment]
Note: Prior to this PR all colors are sorted by hue only, i.e. in the above picture, strictly go from left to right. This results in quite stong saturation variations. This PR separtes the 2D hue/sat space in two groups, splitting at sat=0.2, and for each group draw by hue, i.e. go from left to right.

image

**Comment by timhoffm** [comment]
Not going to follow up.
